cluster-autoscaler/estimator/binpacking_estimator_test.go

```diff
@@ -32,7 +32,7 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
-func makePods(cpuPerPod int64, memoryPerPod int64, hostport int32, maxSkew int32, topologySpreadingKey string, podCount int) []*apiv1.Pod {
+func makePodEquivalenceGroup(cpuPerPod int64, memoryPerPod int64, hostport int32, maxSkew int32, topologySpreadingKey string, podCount int) PodEquivalenceGroup {
```

Comment:
I know you're just modifying an existing function, but could this just accept a pod and count to be composed with `BuildTestPod`? https://github.com/kubernetes/autoscaler/blob/7184d23e62367c57b2f0d66203a5d7c7a7eb661e/cluster-autoscaler/utils/test/test_utils.go#L36

For instance: `makePodEquivalenceGroup(BuildTestPod(...), 10)`? Missing params can be added via option pattern.

Reply:
kept the `BuildTestPod` hidden as a call inside in `makePodEquivalenceGroup`
But this simplifies the code quite.

Reply:
I think with 6 unnamed variables, mostly of the same type, it is hard to read test cases without referring back here. Hence my proposal to explicitly call `BuildTestPod` there - the options are named, is it is clear what they do.

Reply:
Updated